src/backend/columnar/cstore_tableam.c

```diff
@@ -570,39 +537,27 @@ cstore_relation_set_new_filenode(Relation rel,
 								 TransactionId *freezeXid,
 								 MultiXactId *minmulti)
 {
-	DataFileMetadata *metadata = ReadDataFileMetadata(rel->rd_node.relNode, true);
-	uint64 blockRowCount = 0;
-	uint64 stripeRowCount = 0;
-	CompressionType compression = 0;
 	Oid oldRelfilenode = rel->rd_node.relNode;
 
 	MarkRelfilenodeDropped(oldRelfilenode, GetCurrentSubTransactionId());
 
-	if (metadata != NULL)
-	{
-		/* existing table (e.g. TRUNCATE), use existing blockRowCount */
-		blockRowCount = metadata->blockRowCount;
-		stripeRowCount = metadata->stripeRowCount;
-		compression = metadata->compression;
-	}
-	else
-	{
-		/* new table, use options */
-		CStoreOptions *options = CStoreTableAMDefaultOptions();
-		blockRowCount = options->blockRowCount;
-		stripeRowCount = options->stripeRowCount;
-		compression = options->compressionType;
-	}
-
 	/* delete old relfilenode metadata */
 	DeleteDataFileMetadataRowIfExists(rel->rd_node.relNode);
 
 	Assert(persistence == RELPERSISTENCE_PERMANENT);
 	*freezeXid = RecentXmin;
 	*minmulti = GetOldestMultiXactId();
 	SMgrRelation srel = RelationCreateStorage(*newrnode, persistence);
-	InitCStoreDataFileMetadata(newrnode->relNode, blockRowCount, stripeRowCount,
-							   compression);
+	InitCStoreDataFileMetadata(newrnode->relNode);
+
+	/* populate with default options */
+	ColumnarOptions defaultOptions = {
+		.blockRowCount = cstore_block_row_count,
+		.stripeRowCount = cstore_stripe_row_count,
+		.compressionType = cstore_compression
+	};
+	SetColumnarOptions(rel->rd_id, &defaultOptions);
```

Comment:
I think `cstore_relation_set_new_filenode` is also called by `TRUNCATE`, in which case we shouldn't change the table options.